FirebaseMessaging/Sources/FIRMessagingContextManagerService.m

```diff
@@ -129,7 +136,70 @@ + (BOOL)handleContextManagerLocalTimeMessage:(NSDictionary *)message {
   return YES;
 }
 
++ (void)scheduleiOS10LocalNotificationForMessage:(NSDictionary *)message atDate:(NSDate *)date {
+  NSCalendar *calendar = [NSCalendar currentCalendar];
+  if (@available(macOS 10.14, iOS 10.0, watchOS 3.0, tvOS 10.0, *)) {
+    NSCalendarUnit unit = NSCalendarUnitYear | NSCalendarUnitMonth | NSCalendarUnitDay |
+                          NSCalendarUnitHour | NSCalendarUnitMinute | NSCalendarUnitSecond;
+    NSDateComponents *dateComponents = [calendar components:(NSCalendarUnit)unit fromDate:date];
+    UNCalendarNotificationTrigger *trigger =
+        [UNCalendarNotificationTrigger triggerWithDateMatchingComponents:dateComponents
+                                                                 repeats:YES];
```

Comment:
I wonder why `repeats` is `YES`? It looks like it won't repeat any time soon with the specified calendar units (maybe in the next era maybe :) ).

Reply:
Good catch. I think this should be no as backend every day will send a new scheduled notification. And client has no way to set when to end, so this should be a backend control thing. I will double test again today and tomorrow.